radis/lbl/broadening.py

```diff
@@ -902,124 +1065,138 @@ def _calc_broadening_HWHM(self):
-                    #     message=f"{msg}\nYou can silence this error by using `warnings['MissingDiluentBroadeningWarning']='ignore'`.\nThe broadening coefficient of air is used instead.",
-                    #     category="MissingDiluentBroadeningWarning",
+                    #     message="f{msg}\nIf not, you can silence this error by using `warnings['MissingDiluentBroadeningTdepWarning']='ignore'`.\nThe temperature-dependance coefficient of air is used instead.",
+                    #     category="MissingDiluentBroadeningTdepWarning",
                     # )
                     # diluent_broadening_coeff["n_" + diluent_name] = df[
                     #     "Tdpair"
                     # ]  # note @dev : check it doesn't create a new memory object
-            else:
-                msg = f"Temperature dependance of Broadening Coefficient of {self.get_conditions()['molecule']} by {key} not present in database. \nIf the database should include these coefficients, try removing the cache by using once `use_cached='regen'` in calc_spectrum."
-                raise ValueError(
-                    f"{msg}\nOtherwise, replace '{key}' by 'air' in the `diluent=` parameter."
-                )
-                # self.warn(
-                #     message="f{msg}\nIf not, you can silence this error by using `warnings['MissingDiluentBroadeningTdepWarning']='ignore'`.\nThe temperature-dependance coefficient of air is used instead.",
-                #     category="MissingDiluentBroadeningTdepWarning",
-                # )
-                # diluent_broadening_coeff["n_" + diluent_name] = df[
-                #     "Tdpair"
-                # ]  # note @dev : check it doesn't create a new memory object
 
         # Get broadenings
+        # Adds hwhm_lorentz:
+        self._add_Lorentzian_broadening_HWHM(
+            df,
+            pressure_atm,
+            mole_fraction,
+            Tgas,
+            Tref,
+            diluent,
+            diluent_broadening_coeff,
+            isneutral,
+        )
+        # Add hwhm_gauss:
+        self._add_doppler_broadening_HWHM(df, Tgas)
         if broadening_method == "voigt":
-            # Adds hwhm_voigt, hwhm_gauss, hwhm_lorentz:
-            self._add_voigt_broadening_HWHM(
-                df,
-                pressure_atm,
-                mole_fraction,
-                Tgas,
-                Tref,
-                diluent,
-                diluent_broadening_coeff,
-            )
-        elif broadening_method in ["convolve", "fft"]:
-            # Adds hwhm_lorentz:
-            self._add_collisional_broadening_HWHM(
-                df,
-                pressure_atm,
-                mole_fraction,
-                Tgas,
-                Tref,
-                diluent,
-                diluent_broadening_coeff,
+            # Adds hwhm_voigt:
+            df["hwhm_voigt"] = (
+                olivero_1977(2 * df["hwhm_gauss"], 2 * df["hwhm_lorentz"]) / 2
```

Comment:
seems like the function `_add_voigt_broadening_HWHM` is not used anymore in RADIS. Not saying it's good or bad, but we can discuss it.

Reply:
Sorted during meeting. @code29563 wrote the code in a more compact way